modin/pandas/dataframe.py

```diff
@@ -1132,7 +1132,14 @@ def insert(self, loc, column, value, allow_duplicates=False):  # noqa: PR01, D20
                     f"Wrong number of items passed {len(value.columns)}, placement implies 1"
                 )
             value = value.squeeze(axis=1)
-
+        elif isinstance(value, np.ndarray) and len(value.shape) > 1:
+            if value.shape[1] == 1:
```

Comment:
Is this if statement ever hit? I think its supposed to be preventing the second error you caught:
```python
df.insert(0, np.array([[1], [1]]))
```
from erroring, but the `squeeze` above ensures that that array is flattened, and its shape has only 1 element, so I imagine if it enters the `elif` on 1135, we know to raise a ValueError?